core/src/main/scala/tech/beshu/ror/accesscontrol/utils/FilterUtils.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.utils
+
+import com.google.common.base.Strings
+import tech.beshu.ror.Constants
+import tech.beshu.ror.utils.FilterTransient
+
+object FilterUtils {
+
+  def filterFromHeaderValue(value: String) = {
+    Option(FilterTransient.deserialize(value)) match {
+      case Some(ft) =>
+        val filter = ft.getFilter
+        if (!Strings.isNullOrEmpty(filter)) filter
```

Comment:
No serialisation/deserialisation should be necessary if we apply the filter to the search request in ACL space.